script.js

```diff
@@ -64,59 +64,91 @@ function resetGame() {
     pumpkinMove.pause();
     reaperMove.cancel();
     grimReaperZone.removeEventListener('mouseleave', moveReaper);
+
 }
 
 // code here to activate win modal displaying
 function win() {
     winModal.style.display = "block";
+    clearInterval(timer);
+    timerRunning = false;
     resetGame();
 }
 
 // code here to activate lose modal displaying
 function lose() {
     loseModal.style.display = "block";
+    clearInterval(timer);
+    timerRunning = false;
     resetGame();
 }
 
 winModal.addEventListener('click', (e) => {
     e.stopPropagation();
     winModal.style.display = "none";
+    document.querySelector('#timer p').textContent = '00:45';
     start.addEventListener('mouseleave', runGame);
 });
 
 loseModal.addEventListener('click', (e) => {
     e.stopPropagation();
     loseModal.style.display = "none";
+    document.querySelector('#timer p').textContent = '00:45';
     start.addEventListener('mouseleave', runGame);
 });
 
-//event listener on the start button to make splashscreen disappear
+// event listener on the start button to make splashscreen disappear
 document.querySelector('.startButton').addEventListener('click', () => {
     document.querySelector('#splashMain').style.display = 'none';
     document.querySelector('#gameScreen').style.display = 'block';
     start.addEventListener('mouseleave', runGame);
 })
 
-//listen for collisions with obstacles
+// listen for collisions with obstacles
 function listenForCollisions() {
     collision.forEach(item => {
         item.addEventListener('mouseenter', lose);
     });
 }
 
-//win if u get to Exit door (no timers yet)
+// win if you get to Exit door
 function listenForWinning() {
     winningDoor.addEventListener('mouseenter', win);
 }
 function moveReaper() {
     reaperMove.play();
 }
 
-//call main function to start game from here
+let timerRunning = false;
+
+// call main function to start game from here
 function runGame() {
     ghostMove.play();
     pumpkinMove.play();
+    if (timerRunning === false) {
+        startTimer(44, document.querySelector('#timer p'));
+    }
     listenForCollisions();
     listenForWinning();
     grimReaperZone.addEventListener('mouseleave',moveReaper);
-}
\ No newline at end of file
+}
+
+let timer;
+
+function startTimer(duration, element) {
+    timerRunning = true;
+    let timeLeft = duration;
+    timer = setInterval(() => {
+        let minutes = parseInt(timeLeft / 60, 10);
+        let seconds = parseInt(timeLeft % 60, 10);
+
+        minutes = (minutes < 10) ? ("0" + minutes) : minutes;
+        seconds = (seconds < 10) ? ("0" + seconds) : seconds;
+
+        element.textContent = minutes + ":" + seconds;
+
+        if (--timeLeft < 0) {
```

Comment:
generally its not good to manipulate a variable while doing an `if` statement. do the decrement first, then do the comparison